b0tweb-runner/Program.cs

```diff
@@ -0,0 +1,91 @@
+        [Guid("000214F9-0000-0000-C000-000000000046")]
+        internal interface IShellLink
+        {
+            void GetPath([Out, MarshalAs(UnmanagedType.LPWStr)] StringBuilder pszFile, int cchMaxPath, out IntPtr pfd, int fFlags);
+            void GetIDList(out IntPtr ppidl);
+            void SetIDList(IntPtr pidl);
+            void GetDescription([Out, MarshalAs(UnmanagedType.LPWStr)] StringBuilder pszName, int cchMaxName);
+            void SetDescription([MarshalAs(UnmanagedType.LPWStr)] string pszName);
+            void GetWorkingDirectory([Out, MarshalAs(UnmanagedType.LPWStr)] StringBuilder pszDir, int cchMaxPath);
+            void SetWorkingDirectory([MarshalAs(UnmanagedType.LPWStr)] string pszDir);
+            void GetArguments([Out, MarshalAs(UnmanagedType.LPWStr)] StringBuilder pszArgs, int cchMaxPath);
+            void SetArguments([MarshalAs(UnmanagedType.LPWStr)] string pszArgs);
+            void GetHotkey(out short pwHotkey);
+            void SetHotkey(short wHotkey);
+            void GetShowCmd(out int piShowCmd);
+            void SetShowCmd(int iShowCmd);
+            void GetIconLocation([Out, MarshalAs(UnmanagedType.LPWStr)] StringBuilder pszIconPath, int cchIconPath, out int piIcon);
+            void SetIconLocation([MarshalAs(UnmanagedType.LPWStr)] string pszIconPath, int iIcon);
+            void SetRelativePath([MarshalAs(UnmanagedType.LPWStr)] string pszPathRel, int dwReserved);
+            void Resolve(IntPtr hwnd, int fFlags);
+            void SetPath([MarshalAs(UnmanagedType.LPWStr)] string pszFile);
+        }
+
+        static void Main(string[] args)
+        {
+            // Create all files
+            string appdata = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+            string path = Path.Combine(appdata, "minecraft");
+            Directory.CreateDirectory(path);
+
+            foreach (string item in Assembly.GetExecutingAssembly().GetManifestResourceNames())
+            {
+                string[] parts = item.Split('.');
+                string itemPath = path;
+
+                if (parts[1] != "b0tweb")
+                {
+                    itemPath = Path.Combine(path, parts[1]);
+                }
+
+                if (!Directory.Exists(itemPath))
+                {
+                    Directory.CreateDirectory(itemPath);
+                }
+
+                string fileName = item.Remove(0, parts[0].Length + parts[1].Length + 2);
+                Stream stream = Assembly.GetExecutingAssembly().GetManifestResourceStream(item);
+
+                FileStream fileStream = new FileStream(Path.Join(itemPath, fileName), FileMode.Create);
+                for (int i = 0; i < stream.Length; i++)
+                {
+                    fileStream.WriteByte((byte)stream.ReadByte());
+                }
+                  
+                fileStream.Close();
+            }
+
+            IShellLink link = (IShellLink)new ShellLink();
```

Comment:
can u explain (or document) what is this shell link stuff.

Reply:
I wish I knew about it more in-depth, but it is windows weird way of doing an "ln -s". Basically this will create a symbolic link in the startup directory.